elastic/kibana
# [Stack Monitoring] Ensure GlobalState class has it's destroy() method called on unmount

## Summary

This fixes #135136 (and other related problems, e.g. https://github.com/elastic/kibana/issues/139231#issuecomment-1230820859).

Whilst the `GlobalState` class (which syncs to the URL) had a `destroy()` method, it wasn't being called. This is now called on unmount of the top-level `<GlobalStateProvider />`.

## Testing

I was able to consistently replicate the bug by doing the following:

1. Navigate to Discover
2. Navigate to Stack Monitoring
3. Navigate back to Discover
4. Interact with the "Refresh every" toggle of the timepicker (either turn it off or on, it's the change of the refresh that matters)

This should throw a routing error without this PR.
